EIPS/eip-6821.md

```diff
@@ -0,0 +1,41 @@
+---
+eip: 6821
+title: Support ENS Name for Web3 URL
+description: A mapping from an ENS name to the contract address in Web3 URL
+author: Qi Zhou (@qizhou), Qiang Zhu (@qzhodl)
+discussions-to: https://ethereum-magicians.org/t/eip-6821-support-ens-name-for-web3-url/13654
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-04-02
+requires: 137, 634, 4804
+---
+
+## Abstract
+
+This standard defines the mapping from an name of an Ethereum name service (ENS) to an Ethereum address for [ERC-4804](./eip-4804.md).
+
+## Motivation
+
+ERC-4804 defines a `web3://`-scheme RFC 2396 URI to call a smart contract either by its address or a **name** from name service.  If a **name** is specified, the standard specifies a way to resolve the contract address from the name.
+
+## Specification
+
+Given **contractName** and **chainid** from a `web3://` URI defined in ERC-4804, the protocol will find the address of the contract using the following steps:
+
+1. Find the `contentcontract` text record on ENS resolver on chain **chainid**.  Return error if the chain does not have ENS or the record is an invalid ETH address.
+2. If the `contentcontract` text record does not exist, the protocol will use the ETH address of **name**.
```

Comment:
What does "use the ETH address of **name**" mean? The resolved address from [ERC-137's contract address interface](https://eips.ethereum.org/EIPS/eip-137#contract-address-interface)?

Reply:
Yes, it is the contract address in ERC-137.

Reply:
I have updated the text here according to your comments. Please take a look!